lib/paratii.eth.vouchers.js

```diff
@@ -0,0 +1,63 @@
+import { getInfoFromLogs } from './utils.js'
+let dopts = require('default-options')
+
+export class ParatiiEthVouchers {
+  constructor (context) {
+    // context is a ParatiiEth instance
+    this.eth = context
+  }
+
+  async getVouchersContract () {
+    let contract = await this.eth.getContract('Vouchers')
+    if (contract.options.address === '0x0') {
+      throw Error('There is not Vouchers contract known in the registry')
+    }
+    return contract
+  }
+
+  async add (options) {
+    let defaults = {
+      voucherCode: null,
+      amount: undefined
+    }
+    options = dopts(options, defaults)
+
+    if (options.voucherCode === null) {

```

Comment:
The type checking is very much appreciated and useful.

in this particular case, though (checking that arguments are provided), this can be handled by the `default-options` package (setting `voucherCode: undefined` above should handle this case. 

"default-options", which we use, is not very expressive, though - there is another issue open here: https://github.com/Paratii-Video/paratii-lib/issues/50 regarding that.